### PR TITLE
API to get last and current results.

### PR DESCRIPTION
**Pull request by nickwph**
Added these

Last (not including running job):
http://localhost:7777/jobs/1/results/**last**

Current (can be a running job):
http://localhost:7777/jobs/1/results/**current**

### CODE REVIEW

File: models/Result.js [hunk 1]
```diff
@@ -58,7 +58,29 @@ module.exports.init = function (db) {
...
+					.where('job_id', job_id)
+					.where('number', number)
+					.run({}, callback);
+			}
+		},
+		getLast: function (job_id, callback) {
+			Result.findOne()
+				.where('job_id', job_id)
+				.sort('-number')
+				.where('status').ne('running')
```
Comment (thekushpatel) [inline models/Result.js]: So the idea here is to only get completed jobs right? So maybe add also where status does not equal 'queued'.

File: models/Result.js [hunk 2]
```diff
@@ -58,7 +58,29 @@ module.exports.init = function (db) {
...
+				.run({}, callback);
+		},
+		getCurrent: function (job_id, callback) {
+			Result.findOne()
+				.where('job_id', job_id)
+				.sort('-number')
+				.run({}, callback);
 		}
```
Comment (thekushpatel) [inline models/Result.js:84]: Just a thought: should we explicitly put the '/jobs/1/results/last' and '/jobs/1/results/current' as routes? I'm thinking that would make it a bit more self-documenting. Idk, what do you think?

Comment (nickwph) [inline models/Result.js:84]: sure!
i did it this way just because if i add routes there maybe many boilerplate code.

File: models/Result.js (hunk 1 above)
Comment (nickwph) [inline models/Result.js]: okay!

Comment (nickwph) [inline models/Result.js]: i tried putting `.where('status').ne('running').ne('queued')` but it will return the running task

Comment (thekushpatel) [inline models/Result.js]: hmm try `.where('status').ne('running').where('status').ne('queued')`. there might be a better way to do that but it should work fine.

File: models/Result.js (hunk 2 above)
Comment (thekushpatel) [inline models/Result.js:84]: very true, okay makes sense! i like this too.

Comment (nickwph) [inline models/Result.js:84]: oh i just updated the code, shall i roll it back?

File: models/Result.js (hunk 1 above)
Comment (nickwph) [inline models/Result.js]: tried too.

```
.where('status').ne('running')
.where('status').ne('queued') // overriding the last one and will return a running result.
```

File: models/Result.js (hunk 2 above)
Comment (nickwph) [inline models/Result.js:84]: please take a look and see if you like the new way.

Comment (thekushpatel) [inline models/Result.js:84]: If you don't mind. Now that I see it implemented, your first solution was much cleaner! nice work.

File: models/Result.js (hunk 1 above)
Comment (nickwph) [inline models/Result.js]: http://stackoverflow.com/questions/35511823/nor-queries-with-camintejs

File: models/Result.js (hunk 2 above)
Comment (nickwph) [inline models/Result.js:84]: rolled back.

File: models/Result.js (hunk 1 above)
Comment (thekushpatel) [inline models/Result.js]: Try `.nin('status', ['running', 'queued'])`

https://github.com/biggora/caminte/wiki/Query-Interface#nin

Comment (nickwph) [inline models/Result.js]: tried... not working

Comment (nickwph) [inline models/Result.js]: it is still returning a running job

**Comment by nickwph** [comment]
@thekushpatel for some reason the code `.where('status').nin(['queued', 'running'])` works now

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by nickwph** [comment]
@thekushpatel @BruceBGordon please merge!

**Pull request merged**